MiniGameAPI/src/main/java/net/juligames/core/minigame/api/team/procedures/SwitchingInsertionProcedure.java

```diff
@@ -24,10 +24,16 @@ public final class SwitchingInsertionProcedure implements InsertionProcedure {
     @Contract(pure = true)
     @Override
     public @NotNull Boolean apply(@NotNull Set<Team> teams, UUID uuid) {
+        if (lastTeam == null) {
+            teams.stream().findAny().ifPresentOrElse(team -> lastTeam = team,
+                    () -> {
+                        throw new IllegalArgumentException("lastTeam is null and there is no team available to be selected!");
+                    });
+        }
         if (!teams.contains(lastTeam)) {
             throw new IllegalArgumentException("lastTeam of this SwitchingInsertionProcedure is not part of teams!");
         }
-        return new ComparatorInsertionProcedure((o1, o2) -> { //needs testing - maybe reversed order
+        return new ComparatorInsertionProcedure((o1, o2) -> { //needs testing - maybe reversed order - not running - no feedback on what team was selected...
```

Comment:
This file could be deleted from the repository.

But this would prevent the opportunity to use this Procedure with the expectation that it will be fixed in 1.6! 

So a change in the JavaDoc for the class should advise the developers about the potential and the dangers that come with this Procedure. It needs to be clear that the current implementation is not working as advertised!

Reply:
#29